source/common/network/address_impl.cc

```diff
@@ -91,7 +83,7 @@ Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss,
     return std::make_shared<Address::PipeInstance>(sun, ss_len);
   }
   default:
-    throw EnvoyException(fmt::format("Unexpected sockaddr family: {}", ss.ss_family));
+    return nullptr;
```

Comment:
We can keep the error status as a member of Instance and retrieve it later.

Reply:
Or can we make the Instance class loggable and use ENVOY_LOG?

Reply:
I think StatusOr is more natural than either of those approaches.

ENVOY_LOG doesn't have the same functionality -- with the exception we are propagating the detail up the stack so the caller can figure out how to embed it in an error message.

Modifying a class instance variable raises questions of lifetime and/or thread safety.

What's the drawaback of StatusOr?

Reply:
we need to keep the error message in the Instance class if we want to implement statusOr, because the error come from constructors, and constructors don't return anything.

Reply:
You could make the constructor private, and create the object with a static factory method that could do the validation before or after constructing the object.

In this case we'd just return the StatusOr construct that captures `fmt::format("Unexpected sockaddr family: {}", ss.ss_family)` so we don't drop info. The other cases are slightly different; i'll comment below.

Reply:
I did some refactoring, implement a factory method for each instance class constructor. My implementation doesn't look like a good generic factory pattern, is there any graceful way to do this?

Reply:
I think we're still dropping the error message here, but that will be resolved by returning the `StatusOr` here, and all the way up the stack to where we might previously have been catching an exception.